fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -231,8 +252,51 @@ private boolean shouldClearExistingIndexEntries(@Nonnull IndexState indexState,
         }
     }
 
+    @Nonnull
+    public IndexingBase setFallbackMode() {
+        fallbackMode = true;
+        return this;
+    }
+
+    @Nonnull
+    private CompletableFuture<Void> setIndexingTypeOrThrow() {
+        byte[] stamp = getTypeStamp();
+        final List<Object> additionalLogMessageKeyValues = Arrays.asList(LogMessageKeys.CALLING_METHOD, "setIndexingTypeOrThrow",
+                LogMessageKeys.LOCAL_VERSION, stamp);
+
+        return buildCommitRetryAsync( (store, ignore) -> {
+            Transaction transaction = store.getContext().ensureActive();
+            byte[] stampKey = indexBuildTypeSubspace(store, common.getIndex()).getKey();
+            return transaction.get(stampKey)
+                            .thenApply( bytes -> {
+                                if (bytes == null || fallbackMode) {
+                                    transaction.set(stampKey, stamp);
```

Comment:
The name "fallback" might be ambiguous here. I can see there are more than binary options.
A requested indexing method may or may not be usable.
- If the requested method is not usable (for example, the requested source index is not readable or not single-type or not idempotent), we can throw exception for this now. Maybe you don't have to check here since it will be checked in every transaction anyways. But in the future, people may want to have a option for different handlings.
- If the requested method is usable
  - If it is same as the save one, or there is no saved one. Nice and easy.
  - If different, there can be different handlings decided by an enum `OptionWhenIndexingMethodMismatch`:
    - `USE_OLD_INDEXING_METHOD` (aka "fallback")
    - `USE_NEW_INDEXING_METHOD` (force clearing the index and starting from beginning using the new method)
    - `EXCEPTION` (throw a non-retriable exception)